edgelet/edgelet-docker/src/notary.rs

```diff
@@ -0,0 +1,183 @@
+                "could not delete trust directory {}",
+                trust_dir.display()
+            ))
+            .into());
+        }
+    }
+    fs::create_dir_all(&trust_dir).with_context(|_| {
+        ErrorKind::InitializeNotary(format!(
+            "could not create trust directory {}",
+            trust_dir.display()
+        ))
+    })?;
+
+    // Add https to hostname
+    let input_url_https = format!("https://{}", registry_server_hostname);
+    debug!("URL with https is {}", input_url_https);
+
+    // Create Notary Config.json contents
+    let config_contents = json!({
+      "trust_dir": trust_dir,
+      "remote_server": {
+        "url" : input_url_https,
+      },
+      "trust_pinning": {
+        "ca": {
+          "": path,
+        },
+        "disable_tofu": "true"
+      }
+    });
+    debug!("Config JSON contents {}", config_contents);
+
+    // Generate Notary Config.json path
+    let mut config_file_path = trust_dir;
+    config_file_path.push(r"config.json");
+    debug!("Config file path {}", config_file_path.display());
+
+    // Create Notary config file
+    let file = File::create(&config_file_path).with_context(|_| {
+        ErrorKind::InitializeNotary(format!(
+            "could not create notary config file in {}",
+            config_file_path.display()
+        ))
+    })?;
+    serde_json::to_writer(file, &config_contents).with_context(|_| {
+        ErrorKind::InitializeNotary(format!(
+            "could not create notary config file in {}",
+            config_file_path.display()
+        ))
+    })?;
+    debug!("Config JSON file created successfully");
+    Ok(config_file_path)
+}
+
+pub fn notary_lookup(
+    notary_auth: &str,
+    image_gun: &str,
+    tag: &str,
+    config_path: &str,
+    lock: tokio::sync::lock::LockGuard<BTreeMap<String, String>>,
+) -> impl Future<
+    Item = (
+        String,
+        tokio::sync::lock::LockGuard<BTreeMap<String, String>>,
+    ),
+    Error = Error,
+> {
+    Command::new("notary")
+        .env("NOTARY_AUTH", notary_auth)
+        .arg("lookup")
+        .args(&[image_gun, tag])
+        .args(&["-c", config_path])
+        .output_async()
+        .then(|notary_output| {
+            let notary_output = notary_output.with_context(|_| {
+                ErrorKind::LaunchNotary("could not spawn notary process".to_owned())
+            })?;
+            let notary_output_string =
+                String::from_utf8(notary_output.stdout).with_context(|_| {
+                    ErrorKind::LaunchNotary("could not retrieve notary output as string".to_owned())
+                })?;
+            let split_array: Vec<&str> = notary_output_string.split_whitespace().collect();
+            if split_array.len() < 2 {
+                return Err(ErrorKind::LaunchNotary(
+                    "notary digest split array is empty".to_owned(),
+                )
+                .into());
+            }
+            Ok((split_array[1].to_string(), lock))
```

Comment:
Can we document the notary output here? If we had a deserializer implemented for the output, this will be easier to add a test for too. @arsing - thoughts?